### PR TITLE
Add support for local devices and remote appium server

### PR DESCRIPTION
**Pull request by lorjala**
- Local devices are supported with pytest-lockable
- Remote Appium server can be used by passing API URL

**resources.json:**
```
{
    "id": "1",
    "type": "Phone",
    "platform": "Android",
    "online": true,
    "hostname": <HOSTNAME>,
    "model": "Galaxy S21 5G",
    "version": "12"
}
```

### Usage
**Local device (default appium server address):**
```
$ appium # start server on a separate console
...
$ pytest --phone_requirements 'platform=Android&version=12' ...
```
**Local device:**
```
$ appium -a 127.0.0.1 -p 23456 # start server on a separate console
...
$ pytest --phone_requirements 'platform=Android' --appium_server 'http://localhost:23456' ...
```
**STF device:**
```
$ pytest --stf_host <HOST> --stf_token <TOKEN> --phone_requirements 'platform=Android' ...
```

### TODO:

- [x] Examples
- [x] Tests

### CODE REVIEW

File: pytest_stf/plugin.py [hunk 1]
```diff
@@ -87,27 +92,35 @@ def fixture_allocated_phone(pytestconfig):
...
+
+    if hasattr(pytestconfig, '_openstf'):
+        stf = pytestconfig._openstf  # pylint: disable=protected-access
+        stf  # type=StfClient
+        timeout = pytestconfig.getoption('stf_allocation_timeout')
+        requirements = parse_requirements(requirements)
 
-    with stf.allocation_context(requirements, timeout_seconds=timeout) as device:
-        yield device
+        with stf.allocation_context(requirements, timeout_seconds=timeout) as device:
```
Comment (jupe) [inline pytest_stf/plugin.py:105]: timeout was actual allocation timeout when stf releases device internally unless client does it first. need to clarify timeout arg.

File: pytest_stf/plugin.py [hunk 2]
```diff
@@ -87,47 +99,77 @@ def fixture_allocated_phone(pytestconfig):
...
     """
     adb, phone = phone_with_adb
-    with AppiumServer(appium_args) as appium:
-        yield appium, adb, phone
+    if adb:
+        with AppiumServer(appium_args) as appium:
+            yield appium, adb, phone
+    else:
+        appium_api_path = phone.get(
+            'appium_server') or pytestconfig.getoption('appium_server')
```
Comment (jupe) [inline pytest_stf/plugin.py:158]: phone:appium_server was not mentioned in pr description/help, right?

**Review by jupe — APPROVED** [review]
looks good 👍

File: pytest_stf/plugin.py (hunk 2 above)
Comment (lorjala) [reply pytest_stf/plugin.py:158]: added mention of it to README

File: pytest_stf/plugin.py [hunk 3]
```diff
@@ -103,16 +103,19 @@ def fixture_allocated_phone(pytestconfig, lockable):
     assert requirements, 'phone_requirements required'
     assert 'platform' in requirements, 'platform required'
 
+    allocation_timeout = pytestconfig.getoption('allocation_timeout') or 0
```
Comment (jupe) [inline pytest_stf/plugin.py:106]: default value in appium-stf-client is 60s: https://github.com/OpenTMI/stf-appium-python-client/blob/6d568611a7167eaa3629dffab006b5799e025859/stf_appium_client/StfClient.py#L242 . perhaps here should be the same. py-lockable default timeout seems to be longer, 1000s: https://github.com/jupe/py-lockable/blob/c62662a8a5b9410c7abe5622b3e3513f15c6e414/lockable/lockable.py#L19. 
basically those should be the same to be optimal. propose to use `DEFAULT_TIMEOUT` directly from lockable. Please double check these... :)

Comment (lorjala) [reply pytest_stf/plugin.py:106]: pytest-lockable actually sets allocation_timeout to 10 by default so it never even reaches the default value.

If we want to use `DEFAULT_TIMEOUT` from lockable we would have to override it.

**Pull request merged**